### DISCUSSION TITLE
useQuery needs invalidate feature

### DISCUSSION THREAD

**Discussion by LevanKvirkvelia** [TanStack/query#5951] [Ideas]
Hello maintainers,

While working with the useQuery hook, I've identified a recurring need to invalidate a query. The current method of passing the query key to a refresher isn't convenient. Given that the refetch method does not adhere to the enabled behavior, an explicit invalidate method would be ideal.

To clarify, let's consider a scenario:

```
function ParentComponent() {
  const query = useMyComplicatedQuery();

  return (
    <ChildComponent onRefresh={() => {
      queryInfo.invalidate();
    }} />
  );
}
```

**Comment by TkDodo** [comment]
if you have access to the result from `useQuery`, you can use `refetch` returned from it.

**Reply by LevanKvirkvelia** [reply]
Doesn't ```refetch``` ignore ```enabled: false```?

**Reply by TkDodo** [reply]
it does, but why is that a problem? Maybe I don't understand the use-case, but if you render a query and allow it to be refetched, my guess is it wouldn't be disabled ?

**Reply by LevanKvirkvelia** [reply]
in my usecase, i want to queue refetch while fetching a new api token
